src/Infrastructure/Processor/Process/MediaTranslationShopware6ExportProcess.php

```diff
@@ -0,0 +1,150 @@
+use Ergonode\ExporterShopware6\Infrastructure\Model\Shopware6Media;
+use Ergonode\Multimedia\Domain\Entity\AbstractMultimedia;
+use GuzzleHttp\Exception\ClientException;
+use GuzzleHttp\Exception\GuzzleException;
+use Psr\Log\LoggerInterface;
+use Throwable;
+use Webmozart\Assert\Assert;
+
+class MediaTranslationShopware6ExportProcess
+{
+    private Shopware6Connector $connector;
+
+    protected ProductRepositoryInterface $shopwareProductRepository;
+
+    private LanguageRepositoryInterface $languageRepository;
+
+    private ExportRepositoryInterface $exportRepository;
+
+    private LoggerInterface $logger;
+
+    public function __construct(
+        Shopware6Connector $connector,
+        LanguageRepositoryInterface $languageRepository,
+        ExportRepositoryInterface $exportRepository,
+        ProductRepositoryInterface $shopwareProductRepository,
+        LoggerInterface $logger
+    ) {
+        $this->connector = $connector;
+        $this->languageRepository = $languageRepository;
+        $this->exportRepository = $exportRepository;
+        $this->shopwareProductRepository = $shopwareProductRepository;
+        $this->logger = $logger;
+    }
+
+    public function process(
+        Export $export,
+        Shopware6Channel $channel,
+        AbstractMultimedia $multimedia,
+        string $shopwareId
+    ): void {
+        try {
+            $this->ensureMediaTranslations($channel, $multimedia, $shopwareId);
+        } catch (Throwable $exception) {
+            $this->exportRepository->addError($export->getId(), $exception->getMessage(), []);
+        }
+    }
+
+    /**
+     * @throws GuzzleException
+     */
+    private function ensureMediaTranslations(
+        Shopware6Channel $channel,
+        AbstractMultimedia $multimedia,
+        string $shopwareId
+    ): void {
+        //apply translations from shopware
+        $shopware6Media = new Shopware6Media($shopwareId, null);
+        $shopware6Media->setTranslations($this->getMediaTranslations($channel, $shopwareId));
+        //apply translations from ergo
+        $altTranslations = $multimedia->getAlt()->getTranslations();
+        $titleTranslations = $multimedia->getTitle()->getTranslations();
+        $languages = array_unique(array_merge(array_keys($altTranslations), array_keys($titleTranslations)));
+        $ergoTranslations = [];
+        foreach ($channel->getLanguages() as $channelLanguage) {
+            if (!in_array($channelLanguage->getCode(), $languages)) {
+                continue;
+            }
+            if ($this->languageRepository->exists($channel->getId(), $channelLanguage->getCode())) {
+                $shopwareLanguage = $this->languageRepository->load($channel->getId(), $channelLanguage->getCode());
+                Assert::notNull(
+                    $shopwareLanguage,
+                    sprintf('Expected a value other than null for product lang  %s', $channelLanguage->getCode())
```

Comment:
Can we change the part `product lang` to `media update lang` or similar. It's for exception to be more specific so we know which part of app it's thrown

Reply:
Done